go-kit/metrics/provider/librato/batcher.go

```diff
@@ -0,0 +1,424 @@
+package librato
+
+import (
+	"bytes"
+	"encoding/json"
+	"math"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	oldBatcherPath = "/v1/metrics"
+	tagBatcherPath = "/v1/measurements"
+)
+
+type batcher interface {
+	Batch(URL *url.URL, interval time.Duration) ([]*http.Request, error)
+}
+
+type oldBatcher struct {
+	p *Provider
+}
+
+// Batch will batch up all the metrics into []*http.Requests using the old style API.
+func (b *oldBatcher) Batch(u *url.URL, interval time.Duration) ([]*http.Request, error) {
+	// Calculate the sample time.
+	st := time.Now().Truncate(interval).Unix()
+
+	// Sample the metrics.
+	gauges := b.sample(int(interval.Seconds()))
```

Comment:
given my earlier comment, maybe this can turn into:

``` go
measurements := b.sample(int(interval.Seconds()))
gauges = make([]gauge, 0, len(measurements))
for _, m := range measurements {
  gauges = append(gauges, m.Gauge())
}
```